araproc/analysis/standard_reco.py

```diff
@@ -330,3 +384,187 @@ def lookup_arrival_time(self,
+        The theta range is determined based on the average antenna z-coordinate and a threshold.
+
+        Parameters
+        ----------
+        corr_map : dict
+            Dictionary containing:
+            - 'map': ROOT.TH2D histogram with correlation values mapped by theta and phi.
+            - 'corr': Overall peak correlation value.
+            - 'radius': Correlation radius for the station (in meters).
+        z_thresh : float, optional
+            Depth threshold for defining the theta range lower bound. Default is -10 m.
+
+        Returns
+        -------
+        max_surf_corr : float
+            Maximum surface correlation within the theta range.
+        max_theta : float
+            Theta angle corresponding to max surface correlation.
+        max_phi : float
+            Phi angle corresponding to max surface correlation.
+        """
+        # Extract the radius and validate
+        radius = corr_map.get("radius", None)
+        if radius is None:
+            raise ValueError("Radius not found in corr_map.")
+        radius = float(radius)
+
+        # Get antenna z-coordinates and calculate average
+        geom_tool = ROOT.AraGeomTool.Instance()
+        z_coords = [geom_tool.getStationInfo(self.station_id).getAntennaInfo(ant).antLocation[2] 
+                    for ant in range(self.num_channels)]
+        avg_z = sum(z_coords) / len(z_coords)
+
+        # Check if surface is visible at the given radius
+        if radius < (abs(avg_z) + z_thresh):
+            raise RuntimeError("Surface not visible: radius is smaller than average antenna z-coordinate.")
+
+        # Define theta range
+        theta_surface = math.degrees(math.asin(abs(avg_z) / radius))
+        theta_thresh = math.degrees(math.asin((abs(avg_z) + z_thresh) / radius))
+        theta_min, theta_max = min(theta_surface, theta_thresh), max(theta_surface, theta_thresh)
+
+        # Access correlation map and filter values within the theta range
+        hist = corr_map.get("map", None)
+        if hist is None:
+            raise ValueError("The 'map' key was not found in corr_map.")
+        
+        surf_corr_values = []
+        corr_bins = []
+
+        # Loop through the bins and store values within theta range
+        for x_bin in range(1, hist.GetNbinsX() + 1):
+            for y_bin in range(1, hist.GetNbinsY() + 1):
+                theta = hist.GetYaxis().GetBinCenter(y_bin)
+                phi = hist.GetXaxis().GetBinCenter(x_bin)
+                corr = hist.GetBinContent(x_bin, y_bin)
+                if theta_min <= theta <= theta_max:
+                    surf_corr_values.append(corr)
+                    corr_bins.append((x_bin, y_bin))
+
+        # Identify maximum surface correlation
+        if not surf_corr_values:
+            raise RuntimeError("No correlation values found in the specified theta range.")
+        
+        max_surf_corr = np.nanmax(surf_corr_values)
```

Comment:
Let's drop the `np.nan*` functions throughout, so this is just `np.max`. If there's a `nan` I think we want there to be an error or something.

Reply:
Done. Added a ValueError to catch nans.